dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layerparam/initializer/PoolingLayerParameterInitializer.java

```diff
@@ -86,8 +92,22 @@ public int getIndex() {
     if (inputShape.length != 2) {
       throw new IllegalArgumentException("Unsupported input dimensions: " + inputShape.length);
     }
-    computedShape[0] = (int) Math.ceil((float) (inputShape[0] - kernelHeight) / strideHeight) + 1;
-    computedShape[1] = (int) Math.ceil((float) (inputShape[1] - kernelWidth) / strideWidth) + 1;
+    if (paddingHeight >= kernelHeight) {
+      throw new IllegalArgumentException("Padding height should be less than kernel height.");
+    }
+    if (paddingWidth >= kernelWidth) {
+      throw new IllegalArgumentException("Padding width should be less than kernel width.");
+    }
+    computedShape[0] = (int) Math.ceil((float) (inputShape[0] - kernelHeight + 2 * paddingHeight) / strideHeight) + 1;
+    computedShape[1] = (int) Math.ceil((float) (inputShape[1] - kernelWidth + 2 * paddingWidth) / strideWidth) + 1;
+    //Pooling should start inside the input images.
+    //If the last pooling starts outside the input image, clip that output.
+    if ((computedShape[0] - 1) * strideHeight >= inputShape[0] + paddingHeight) {
+      --computedShape[0];
+    }
+    if ((computedShape[1] - 1) * strideWidth >= inputShape[1] + paddingWidth) {
+      --computedShape[1];
+    }
```

Comment:
Shouldn't we double check that `(computeShape[0] - 1) * strideHeight < inputShape[0] + paddingHeight` and `(computedShape[1] - 1) * strideWidth < inputShape[1] + paddingWidth)`?
Caffe do this.